dapr_agents/agents/components.py

```diff
@@ -67,38 +98,80 @@ def __init__(
+                    and component.name == "agent-wfstatestore"
+                    and state is None
+                ):
+                    state = AgentStateConfig(
+                        store=StateStoreService(store_name=component.name),
+                        state_key=f"{name.replace(' ', '-').lower() if name else 'default'}:workflow_state",
+                    )
+                if (
+                    "state" in component.type
+                    and component.name == "agent-registry"
+                    and registry is None
+                ):
+                    registry = AgentRegistryConfig(
+                        store=StateStoreService(store_name="agent-registry"),
+                        team_name="default",
+                    )
+                if "state" in component.type and component.name == "agent-runtime":
+                    raw_runtime_conf: StateResponse = _client.get_state(
+                        store_name=component.name,
+                        key="agent_runtime_configuration",
+                    )
+                    try:
+                        self._runtime_conf = (
+                            json.loads(raw_runtime_conf.data)
+                            if raw_runtime_conf.data
+                            else {}
                         )
-                    if component.name == "agent-registry" and registry is None:
-                        registry = AgentRegistryConfig(
-                            store=StateStoreService(store_name="agent-registry"),
-                            team_name="default",
+                        for key, value in self._runtime_conf.items():
+                            logger.info(f"Runtime configuration: {key}={value}")
+                    except json.JSONDecodeError:
+                        logger.warning(
+                            "Failed to decode agent runtime configuration JSON. Using empty configuration."
                         )
-                    if (
-                        "pubsub" in component.type
-                        and component.name == "agent-pubsub"
-                        and pubsub is None
-                    ):
-                        logger.info(f"topic: {name}.topic")
-                        pubsub = AgentPubSubConfig(
-                            pubsub_name="agent-pubsub",
-                            agent_topic=f"{name.replace(' ', '-').lower()}.topic",
-                            broadcast_topic="agents.broadcast",
+                if (
+                    "pubsub" in component.type
+                    and component.name == "agent-pubsub"
+                    and pubsub is None
+                ):
+                    logger.info(f"topic: {name}.topic")
+                    pubsub = AgentPubSubConfig(
+                        pubsub_name="agent-pubsub",
+                        agent_topic=f"{name.replace(' ', '-').lower()}.topic",
+                        broadcast_topic="agents.broadcast",
```

Comment:
what if someone has set agent-pubsub and we're picking it up, but they set a diff broadcast_topic? Same for the state components... then, we would be overriding their value.

Reply:
Actually, shouldn't that be covered by the conditional:
```python
if (
    "pubsub" in component.type
    and component.name == "agent-pubsub"
    and pubsub is None
):
  ...
```

If they've added:

```yaml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: agent-pubsub
spec:
  type: pubsub.redis
  version: v1
  metadata:
    - name: redisHost
      value: localhost:6379
```

But they have not added it in the agent instantiation then this does exactly what we want it to, no?
`"pubsub" in component.type` -> We found the component
`and component.name == "agent-pubsub"` -> It has the right default naming
`and pubsub is None` -> And they didn't pass any during instantiation

Then the agent would pick it up as the default pubsub